File: src/ui/src/models/api/Conversation/GetConversationsItemModel.ts
/**
 * Model for a conversation item returned from the API
 */
export interface GetConversationsItemModel {
  /**
   * The unique identifier for the conversation
   */
  id: string;
  
  /**
   * The title of the conversation
   */
  title: string;
  
  /**
   * The last message in the conversation
   */
  lastMessage: string;
  
  /**
   * The number of messages in the conversation
   */
  messageCount: number;
  
  /**
   * The timestamp when the conversation was created
   */
  createdAt: string;
  
  /**
   * The timestamp when the conversation was last updated
   */
  updatedAt: string;
}

File: src/backend/DonkeyWork.Chat.Api/Models/Conversation/GetConversationsItemModel.cs
// ------------------------------------------------------
// <copyright file="GetConversationsItemModel.cs" company="DonkeyWork.Dev">
// Provided as is by (c) DonkeyWork.Dev. No warranty or support is given.
// </copyright>
// ------------------------------------------------------

namespace DonkeyWork.Chat.Api.Models.Conversation;

/// <summary>
/// A conversation List Model.
/// </summary>
public record GetConversationsItemModel
{
    /// <summary>
    /// Gets the conversation id.
    /// </summary>
    public Guid Id { get; init; }

    /// <summary>
    /// Gets the Title.
    /// </summary>
    public string Title { get; init; } = string.Empty;

    /// <summary>
    /// Gets a portion of the last message.
    /// </summary>
    public string LastMessage { get; init; } = string.Empty;

    /// <summary>
    /// Gets the message Count.
    /// </summary>
    public int MessageCount { get; init; }

    /// <summary>
    /// Gets the Created At.
    /// </summary>
    public DateTimeOffset CreatedAt { get; init; }

    /// <summary>
    /// Gets the Updated At.
    /// </summary>
    public DateTimeOffset UpdatedAt { get; init; }
}